src/PIL/GifImagePlugin.py

```diff
@@ -523,6 +523,8 @@ def _normalize_palette(im, palette, info):
                 index = im.palette.colors[source_color]
             except KeyError:
                 index = None
+            if index in used_palette_colors:
+                index = None
```

Comment:
I think the `if` could be inside the `try`, since if there's an exception it will have already been set to `None`.

Reply:
I actually have a different idea of how to simplify the code. In my latest commit, I replace the `try` block with `get()`.